rama-tcp/src/pool/mod.rs

```diff
@@ -0,0 +1,182 @@
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+
+use rama_core::error::OpaqueError;
+use rand::{seq::SliceRandom, RngCore};
+
+use crate::client::TcpStreamConnector;
+
+/// Selection algorithms
+#[derive(Debug, Clone)]
+pub enum Selector {
+    RoundRobin(Arc<AtomicUsize>),
+    Random,
+}
+
+impl Selector {
+    fn new_random() -> Self {
+        Self::Random
+    }
+
+    fn new_round_robin() -> Self {
+        Self::RoundRobin(Arc::new(AtomicUsize::default()))
+    }
+
+    fn next<C: Clone>(&self, connectors: &[C]) -> Option<C> {
+        if connectors.is_empty() {
+            return None;
+        }
+        let selection = match self {
+            Selector::RoundRobin(atomic_usize) => atomic_usize.fetch_add(1, Ordering::Relaxed),
+            Selector::Random => rand::rng().next_u64() as usize,
```

Comment:
Maybe this should be `next_u32` for 32bit systems?

Reply:
I could do something like this:
```rust
            Selector::Random => {
                #[cfg(target_pointer_width = "64")]
                let selection = rand::rng().next_u64() as usize;
                #[cfg(target_pointer_width = "32")]
                let x = rand::rng().next_u32() as usize;
                selection
            }
```
However `ThreadRng` does not provide a `next_u16`, therefore we would still have the same issue.

Reply:
It's fine to do u64 for now :) There's plenty of the codebase that would potentially break on 32-bit machines. For now I am only aiming for 64 bit machines. It will require a large sponsor (financially or in code contributions and maintenance) to also support 32 bit platforms.

All servers these days are 32 bit machines, and don't think you can still buy a computer, phone or tablet that is still 32 bit. Anyway, not my concern for now. Thank you for bringing it up though.

If we ever need to support such things we probably want to ensure that anything that requires a difference between 32 bit and 64 bit to be in a single module / crate somewhere so we don't need this kind of logic everywhere. For now it is what it is.